walletunlocker/service.go

```diff
@@ -306,9 +342,33 @@ func (u *UnlockerService) InitWallet(ctx context.Context,
 		initMsg.ChanBackups = *chansToRestore
 	}
 
-	u.InitMsgs <- initMsg
+	// Deliver the initialization message back to the main daemon.
+	select {
+	case u.InitMsgs <- initMsg:
+		// If the client instructs us to do a stateless initialization
+		// where no macaroons should be created by the daemon, there's
+		// an additional step to be done.
+		if in.StatelessInit {
+			// We need to read from the channel to let the daemon
+			// continue its work and to get the admin macaroon.
+			// Once the response arrives, we directly forward it to
+			// the client.
+			select {
+			case adminMac := <-initMsg.MacResponseChannel:
+				return &lnrpc.InitWalletResponse{
```

Comment:
could we also return it as part of the response in a stateful init? Less special casing, and it could just be ignored by the client.

Reply:
Yeah, makes sense. Updated.